src/canvas-extractor/canvas_extractor/__main__.py

```diff
@@ -127,14 +130,27 @@ def extract_assignments(
 
 
 def extract_submissions(
-    assignments: List[Assignment], sync_db: sqlalchemy.engine.base.Engine
-) -> List[Submission]:
+    assignments: List[Assignment],
+    sections: List[Section],
+    sync_db: sqlalchemy.engine.base.Engine,
+):
     logger.info("Extracting Submissions from Canvas API")
-    submissions: List[Submission] = request_submissions(assignments)
-    submissions_df: DataFrame = submissions_synced_as_df(submissions, sync_db)
-    # Temporary - just for demonstration until UDM mapping
-    submissions_df.to_csv("data/submissions.csv", index=False)
-    return submissions
+    export: Dict[Tuple[str, str], DataFrame] = {}
+    for section in sections:
+        for assignment in [
+            assignment
+            for assignment in assignments
+            if assignment.course_id == section.course_id
+        ]:
+            submissions: List[Submission] = request_submissions(assignment)
+            submissions_df: DataFrame = submissions_synced_as_df(submissions, sync_db)
+            submissions_df = map_to_udm_submissions(submissions_df)
+            export[(section.id, assignment.id)] = submissions_df
+
+    SUBMISSIONS_OUTPUT_DIRECTORY = os.path.join(
+        BASE_OUTPUT_DIRECTORY, SUBMISSION_ROOT_DIRECTORY
+    )
+    write_multi_tuple_csv(export, datetime.now(), SUBMISSIONS_OUTPUT_DIRECTORY)
 
 
 def extract_enrollments(
```

Comment:
Would it make sense to move any of this code to a facade class?

Reply:
Now that these have multiplied in the main file, I'd suggest moving these functions to their own modules in a similar pattern to the "api" and "mapping" module groupings.  Maybe under an "extract" grouping?

Reply:
I like the idea of a facade